File: src/main/java/com/github/xumeng/java/problems/easy/problem0232/MyQueue0232.java
package com.github.xumeng.java.problems.easy.problem0232;

import java.util.Stack;

public class MyQueue0232 {

    private final Stack<Integer> stack1 = new Stack<>();
    private final Stack<Integer> stack2 = new Stack<>();

    public MyQueue0232() {

    }

    public void push(int x) {
        stack1.push(x);
    }

    public int pop() {
        while (!stack1.isEmpty()) {
            Integer element = stack1.pop();
            if (stack1.isEmpty()) {
                while (!stack2.isEmpty()) {
                    stack1.push(stack2.pop());
                }
                return element;
            }
            stack2.push(element);
        }
        return 0;
    }

    public int peek() {
        while (!stack1.isEmpty()) {
            Integer element = stack1.pop();
            stack2.push(element);
            if (stack1.isEmpty()) {
                while (!stack2.isEmpty()) {
                    stack1.push(stack2.pop());
                }
                return element;
            }
        }
        return 0;
    }

    public boolean empty() {
        return stack1.isEmpty();
    }

}

File: src/main/kotlin/com/github/xumeng/kotlin/problems/easy/problem0232/MyQueue0232.kt
package com.github.xumeng.kotlin.problems.easy.problem0232

class MyQueue0232 {

    private val stack1: ArrayDeque<Int> = ArrayDeque()
    private val stack2: ArrayDeque<Int> = ArrayDeque()

    fun push(x: Int) {
        stack1.addLast(x)
    }

    fun pop(): Int {
        while (stack1.isNotEmpty()) {
            val element = stack1.removeLast()
            if (stack1.isEmpty()) {
                while (stack2.isNotEmpty()) {
                    stack1.addLast(stack2.removeLast())
                }
                return element
            }
            stack2.addLast(element)
        }
        return 0
    }

    fun peek(): Int {
        while (stack1.isNotEmpty()) {
            val element = stack1.removeLast()
            stack2.addLast(element)
            if (stack1.isEmpty()) {
                while (stack2.isNotEmpty()) {
                    stack1.addLast(stack2.removeLast())
                }
                return element
            }
        }
        return 0
    }

    fun empty(): Boolean {
        return stack1.isEmpty()
    }

}